src/dmd/root/file.d

```diff
@@ -63,6 +63,14 @@ struct File
         {
             return buffer.extractData();
         }
+
+        /// ditto
+        /// Include the null-terminator at the end of the buffer in the returned array.
+        ubyte[] extractDataZ() @nogc nothrow pure
+        {
+            auto result = buffer.extractData();
+            return result.ptr[0 .. result.length + 1];
```

Comment:
Interestingly, this isn't covered.
In any case, I think this approach is not the best. Relying on `\0` termination of `char[]` is a bad idea.
I much rather recommend to use `.toCStringThen` when calling a C API function.

Reply:
Note that this is the current behavior, it was just hidden in pointer arithmetic before. toCString doesn't work here since the function actually delivers zero-terminated strings by putting null bytes in a mutable buffer. If the argument is at the end of the buffer, this null byte needs to be there.

Reply:
The buffer ends up being stored in `Strings` right ?

Reply:
Yes, and `Strings` == `Array!(const(char)*)`

Reply:
I see. I originally thought it was solely to call a C API function, my bad.

Reply:
Hum, was working on something with the same needs, and realized this is wrong.
You don't have the guarantee that `extractData` is `\0` terminated.
See https://github.com/dlang/dmd/pull/10310

Reply:
That PR is about OutBuffer, `extractDataZ` only applies to `ReadResult` of `File` which always allocates even 2 zero-terminators:
https://github.com/dlang/dmd/blob/8bbb2aee973fbd85d3e867285e47883d3ed1c18b/src/dmd/root/file.d#L109